sympy/integrals/tests/test_failing_integrals.py

```diff
@@ -287,3 +287,9 @@ def test_issue_4311_slow():
 def test_issue_20370():
     a = symbols('a', positive=True)
     assert integrate((1 + a * cos(x))**-1, (x, 0, 2 * pi)) == (2 * pi / sqrt(1 - a**2))
+
+
+@XFAIL
+def test_pr_23583():
+    # meijerg gives wrong Piecewise with Abs result
+    assert integrate(1/sqrt((x - I)**2-1)) == log(2*x + 2*sqrt(x**2 - 2*I*x - 2) - 2*I)
```

Comment:
I find XFAIL tests like this difficult because the correct result is not necessarily going to be in that form. We don't know that fixing the bug would make this pass.

This is the kind of situation where I would rather have a test that fails as soon as the output of integrate is changed in any way for this case. In other words I would rather have a test that asserts that the output is what it currently is but that is clearly commented to say that the output is wrong and the test should be changed as soon as integrate is fixed. Then if a change causes the test to fail someone can look into it to see if the bug is fixed.

Also if the wrong output currently comes from meijerg it would be good to have separate tests for `integrate` and for `integrate` specifically with `meijerg`.